.github/api_best_practices.md

```diff
@@ -0,0 +1,231 @@
+# ACS API Guidelines
+An overview of the set of standardized guidelines and practices to ensure that the APIs across different 
+projects and teams adhere to a common structure and design. This consistency simplifies the process of understanding 
+and maintaining APIs. The guidelines helps identify potential pitfalls and design choices that could lead to issues to 
+help developers avoid common mistakes and build more robust APIs. When in doubt, It is encouraged to seek resolution
+in slack channel [#forum-acs-api-design](https://redhat-internal.slack.com/archives/C05MMG2PP8A).
+
+## Table of Contents
+
+- [Naming Guidelines](#naming-guidelines)
+  - [gRPC Service Name](#grpc-service-name)
+  - [gRPC Method Name](#grpc-method-name)
+  - [gRPC Message Name](#grpc-message-name)
+
+## Naming Guidelines
+
+### gRPC Service Name
+Service name **must** be unique and should use a noun that generally refers to a resource or product component 
+e.g. `DeploymentService`, `ReportService`, `ComplianceService`. Intuitive and well-known short forms or 
+abbreviations may be used in some cases (and could even be preferable) for succinctness 
+e.g. `ReportConfigService`, `RbacService`. 
+All gRPC methods grouped into a single service must generally pertain to the primary resource of the service.
+
+Note:
```

Comment:
This doesn't seem to fit under the naming section.  Should it instead be its own section?  If there is going to be another PR about API versioning, perhaps this note would fit better in that PR.